test/mock/server.go

```diff
@@ -0,0 +1,253 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+var name string
+var port string
+
+var data map[string]string
+
+func init() {
+	// name of the mock server
+	name = os.Getenv("MOCK_NAME")
+	if len(name) == 0 {
+		panic("This server must have a name, found in MOCK_NAME env")
+	}
+
+	// port of the mock server
+	port = os.Getenv("MOCK_PORT")
+	if len(port) == 0 {
+		port = "8080"
+	}
+
+	// initialize the in memory data
+	data = make(map[string]string)
+}
+
+func suicideHandler(w http.ResponseWriter, r *http.Request) {
+
+	p := mux.Vars(r)
+	delayStr := p["delay"]
+
+	delay, err := strconv.Atoi(delayStr)
+
+	if err != nil {
+		delay = 0
+	}
+
+	fmt.Fprintln(w, "ok")
+
+	f, ok := w.(http.Flusher)
+	if ok && f != nil {
+		f.Flush()
+	}
+
+	timer := time.NewTimer(time.Second * time.Duration(delay))
+	fmt.Printf("%s will autokill in %d second(s)\n", name, delay)
```

Comment:
Since we already have glog in Godeps, I would use glog to make the output consistent and have timestamps.